iroh-p2p/src/rpc.rs

```diff
@@ -252,15 +253,37 @@ impl RpcP2p for P2p {
     }
 
     #[tracing::instrument(skip(self, req))]
-    async fn peer_connect(&self, req: ConnectRequest) -> Result<ConnectResponse> {
+    async fn peer_connect_by_peer_id(&self, req: ConnectByPeerIdRequest) -> Result<()> {
```

Comment:
So connect_by_peer_id will try to figure out the addresses itself, whereas connect requires you to provide addresses?

Reply:
Yes, I'll add some comments to make it more clear.

`connect_by_peer_id` will attempt to find the peer's address via the dht using the `PeerId.

If you `connect` by `Multiaddr` it will just dial the given address.

Reply:
Yes, I tried it out. It even finds my two go-ipfs nodes.